### PR TITLE
Print a number on each card on the board

### PR DESCRIPTION
**Pull request by sorens**
There were a few other changes that became necessary to make it easier to print the card numbers and debug it.

### CODE REVIEW

File: board.py [hunk 1]
```diff
@@ -1,39 +1,122 @@
...
+        print("noble deck:")
+        print(self.noble_deck)
+        for card in self.noble_deck:
+            print(card)
+            d.append(card)
+            if len(d) == 4:
+                NobleRowOfSevenLines(d).print_row()
+                d = []
+        if len(d) > 0:
+            RowOfSevenLines(d).print_row()
```
Comment (yorens) [inline board.py]: should use NobleRowOfSevenLines

File: board.py [hunk 2]
```diff
@@ -121,11 +204,22 @@ def row_6(self):
...
 
     def row_7(self):
+        card_number = 1
+        rank = self.cards[0].get_rank()
+        if rank == 1:
+            card_number = 13
+        elif rank == 2:
+            card_number = 9
+        elif rank == 3:
+            card_number = 5
```
Comment (yorens) [inline board.py]: should not have a rank == 3, and all numbers should be 4 lower than they are now.

File: card.py [hunk 3]
```diff
@@ -9,6 +9,12 @@ def __init__(self, rank, point_value, gem_type, diamond_cost=0, sapphire_cost=0,
         self.ruby_cost = ruby_cost
         self.onyx_cost = onyx_cost
 
+    def __repr__(self):
+        return "{rank: " + str(self.rank) + ", point_value: " + str(self.point_value) + ", gem_type: " + str(self.gem_type) + "}"
+
+    def __str__(self):
+        return 'Card(rank='+str(self.rank)+', point_value='+str(self.point_value)+', gem_type='+str(self.gem_type)+', diamond_cost='+str(self.diamond_cost)+', sapphire_cost='+str(self.sapphire_cost)+', emerald_cost='+str(self.emerald_cost)+', ruby_cost='+str(self.ruby_cost)+', onyx_cost='+str(self.onyx_cost) + ')'
```
Comment (yorens) [inline card.py]: for all of the different types of gems, they should have `yildor.diamond` instead 'diamond' as an example

**Review by yorens — CHANGES_REQUESTED** [review]

**Comment by sorens** [comment]
@yorens I've made the changes based on your feedback!

**Review by yorens — APPROVED** [review]
:P

**Pull request merged**